enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,192 @@
+# Improved Support for Managing Operators through Policy
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+  [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Enhance the policy framework with a new kind, `OperatorPolicy`, to allow for better management of
+OLM Operator installations on managed clusters. The controller for this kind would be included in
+the configuration-policy-controller container, on managed clusters. The new controller would better
+handle imperative installation steps for OLM objects, and provide better support for upgrade and
+uninstallation scenarios.
+
+## Motivation
+
+It is currently possible to install OLM Operators on managed clusters with `ConfigurationPolicies`,
+but it can be complex and error-prone. For example, multiple policies are often needed in order to
+manage the different OLM resources such as OperatorGroup, Subscription, InstallPlan, and
+ClusterServiceVersion. Upgrading or uninstalling an operator can also be difficult with
+ConfigurationPolicies, since it is not generally as easy as just updating one value or deleting a
+resource.
+
+Creating a new policy type will help streamline the process of declaring an operator to be installed
+on a cluster. A new controller will help cover specialized cases that come up when managing operator
+installations, rather than teaching the existing configuration-policy-controller to deal with
+specific types.
+
+### Goals
+
+1. It should be easy to install an operator on a managed cluster with just one policy.
+2. It should be possible to limit updates to operators on clusters, so they do not upgrade beyond a
+   known good version defined in the policy.
+3. The policy managing the operator should clearly report back the status of the installation, with
```

Comment:
Today there are cases where I think something happens in the wrong order and you end up with a failed operator install.  You can't resolve except by doing some cleanup and then trying again.  Does that sound familiar and would that case be covered because the "ordering" is no longer a concern?  Or would there still be a need for a retry on failure?

Reply:
By having the new controller manage the "imperative" steps that could be done in the wrong order, I think it can avoid at least some of the issues we've seen before when we just apply all the resources at once with ConfigurationPolicies.

It might be impossible to prevent all installation errors, but depending on what's required to "cleanup and try again", I think that could also be handled by the controller automatically.